internal/pkg/api/handleAck.go

```diff
@@ -356,18 +360,29 @@ func (ack *AckT) handlePolicyChange(ctx context.Context, zlog zerolog.Logger, ag
 
 func (ack *AckT) updateAPIKey(ctx context.Context,
 	zlog zerolog.Logger,
-	agentID string,
+	agent *model.Agent,
 	currRev, currCoord int64,
-	policyID, apiKeyID, permissionHash string,
+	apiKeyID, permissionHash string,
 	toRetireAPIKeyIDs []model.ToRetireAPIKeyIdsItems) error {
 
 	if apiKeyID != "" {
 		res, err := ack.bulk.APIKeyRead(ctx, apiKeyID, true)
 		if err != nil {
-			zlog.Error().
-				Err(err).
-				Str(LogAPIKeyID, apiKeyID).
-				Msg("Failed to read API Key roles")
+			if isAgentActive(ctx, zlog, ack.bulk, agent.Id) {
+				zlog.Error().
+					Err(err).
+					Str(LogAPIKeyID, apiKeyID).
+					Msg("Failed to read API Key roles")
+			} else {
+				// not an error, race when API key was invalidated before acking
+				zlog.Info().
+					Err(err).
+					Str(LogAPIKeyID, apiKeyID).
+					Msg("Failed to read invalidated API Key roles")
+
+				// update to inactive for future checks
+				agent.Active = false
```

Comment:
This is still an error.  If the ApiKey is gone, the ACK should fail.  We should still output an error, but it should be an info level log message.  Ie. this is an error, but it is expected.

It is also unexpected to have this call produce a side effect of changing the agent record.

Reply:
discussed with @scunningham offline, approach agreed on is: returning error and propagating it to the client while decreasing log level on the server side inside handler